benchmarks/vector_db_compare/minima_treedb_runner.py

```
@@ -198,15 +353,210 @@ def wait_ready(self, expected_count: int | None = None) -> None:
+                    "before_stats": self.controller.stats_snapshot(),
+                    "outcome": "in_progress", "profile_capture": {"status": "not_triggered"},
+                }
+                self.batch_correlations.append(correlation)
+            done = threading.Event()
+            capture_started = threading.Event()
+            capture_lock = threading.Lock()
+
+            def capture(reason: str) -> None:
+                if self.diagnostics_dir is None:
+                    return
+                with capture_lock:
```

Ruff (0.16.2)

[warning] Function definition does not bind loop variable `capture_lock`

(B023)